examples/examples/form_examples.py

```diff
@@ -223,6 +228,71 @@ def field_error_messages(field, **_):
     )
 
 
+@example(gettext("Form children do not need to be all fields"))
+def form_example_children_that_are_not_fields(request):
+    def on_submit(form, **_):
+        if not form.is_valid():
+            return
+        return html.pre(f"You posted: {form.apply(Struct())}").bind(request=request)
+
+    def post_validation(form, **_):
+        if form.fields.f1.value is not None and form.fields.f2.value is not None and form.fields.f3.value is not None:
+            if form.fields.f1.value + form.fields.f2.value != form.fields.f3.value:
+                form.add_error("Calculate again!")
+
+    return Form(
+        iommi_style="bulma",
+        title="Children that are not fields",
+        fields__name=Field(),
+        fields__color=Field.choice(choices=['Red', 'Green', 'Blue']),
+        fields__in_box=html.div(
+            children__f1=Field.integer(attrs__class__column=True),
+            children__plus=html.span("+", attrs__class={'column': True, 'is-narrow': True}),
+            children__f2=Field.integer(attrs__class__column=True,),
+            children__equals=html.span("=", attrs__class={'column': True, 'is-narrow': True}),
+            children__f3=Field.integer(attrs__class_column=True),
+            iommi_style="bulma_query_form",
+            attrs__class={'box': True, 'columns': True, 'is-vcentered': True}
+        ),
+        post_validation=post_validation,
+        actions__submit__post_handler=on_submit
+    )
+
+
+@example(gettext("Form children do not need to be all fields -- declarative"))
+def form_example_children_that_are_not_fields_declarative(request):
+    def on_submit(form, **_):
+        if not form.is_valid():
+            return
+        return html.pre(f"You posted: {form.apply(Struct())}").bind(request=request)
+
+    def post_valid(form, **_):
+        if form.fields.f1.value is not None and form.fields.f2.value is not None and form.fields.f3.value is not None:
+            if form.fields.f1.value + form.fields.f2.value != form.fields.f3.value:
+                form.add_error("Calculate again!")
+
+    class MyForm(Form):
+        name = Field()
+        color = Field.choice(choices=['Red', 'Green', 'Blue'])
+        in_a_box = html.div(
+            children__f1=Field.integer(attrs__class__column=True),
+            children__plus=html.span("+", attrs__class={'column': True, 'is-narrow': True}),
+            children__f2=Field.integer(attrs__class__column=True,),
+            children__equals=html.span("=", attrs__class={'column': True, 'is-narrow': True}),
+            children__f3=Field.integer(attrs__class_column=True),
+            iommi_style="bulma_query_form",
+            attrs__class={'box': True, 'columns': True, 'is-vcentered': True}
+        )
+
+        class Meta:
+            iommi_style = "bulma"
```

Comment:
Why hardcoded style?

Reply:
Cause otherwise the box doesn't look like a box.  Some more documentation needs to be written I guess.

Basically I roughly believe that in our examples we do something that in a real app you would "basically never" do (at least not for any given page of the app).   Support multiple different CSS frameworks within the same app.

So iommi tries hard(ish) to give you default forms that look ok in each CSS framework.  But as a user of iommi, you should not feel like you have to pay that additional complexity tax.  That is for a given page (or more likely for all of your app) you are going to chose one and than it is perfectly fine to hardcode css classes specific to the CSS framework you are using. 

And really we should have a tutorial that explains those nuances.  But writing those is not my super power...

Reply:
Check.